### PR TITLE
feat(generator)!: Template generation for multiple element types

### PR DESCRIPTION
**Pull request by chillleader**
## Description

Support generating multiple element templates for different element types in one run.

## Related issues

related to https://github.com/camunda/team-connectors/issues/547

### CODE REVIEW

File: element-template-generator/congen-cli/src/main/java/io/camunda/connector/generator/cli/command/Generate.java [hunk 1]
```diff
@@ -53,16 +53,21 @@ public Integer call() {
...
       System.err.println("Generation failed: " + e.getMessage());
       return 1;
     }
     try {
-      var resultString = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(template);
-      System.out.println(resultString);
+      for (var template : templates) {
+        var resultString = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(template);
+        System.out.println("---\n");
+        System.out.println(resultString);
```
Comment (sbuettner) [inline element-template-generator/congen-cli/src/main/java/io/camunda/connector/generator/cli/command/Generate.java:69]: Just thinking out loud whether we should render it as an json array to simplify automated processing?

File: element-template-generator/core/src/main/java/io/camunda/connector/generator/java/util/ConfigurationUtil.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  public static GeneratorConfiguration fromAnnotation(
+      ElementTemplate annotation, GeneratorConfiguration override) {
+    var templateId = annotation.id();
+    if (override.templateId() != null) {
+      templateId = override.templateId();
+    }
+    var templateName = annotation.name();
+    if (override.templateName() != null) {
+      templateName = override.templateName();
+    }
```
Comment (markfarkas-camunda) [inline element-template-generator/core/src/main/java/io/camunda/connector/generator/java/util/ConfigurationUtil.java:36]: 🙃nitpicking: Maybe rewrite all of these to a one liner:
`var templateName = Optional.ofNullable(override.templateName()).orElseGet(() -> annotation.name())`

File: element-template-generator/congen-cli/src/main/java/io/camunda/connector/generator/cli/command/ConGen.java [hunk 3]
```diff
@@ -46,11 +51,44 @@ public class ConGen {
...
   GeneratorConfiguration generatorConfiguration() {
+    var bpmnTypes =
+        elementTypes == null
+            ? null
+            : elementTypes.stream()
+                .map(BpmnType::fromName)
+                .map(
+                    bpmnType ->
+                        new ConnectorElementType(getAppliesToFromBpmnType(bpmnType), bpmnType))
+                .collect(Collectors.toSet());
```
Comment (markfarkas-camunda) [inline element-template-generator/congen-cli/src/main/java/io/camunda/connector/generator/cli/command/ConGen.java:71]: 🙃 nitpicking: maybe we can avoid the ternary operator this way:
```
Optional.ofNullable(elementTypes).map(types -> types.stream()
         .map(BpmnType::fromName)
         .map(
                    bpmnType ->
                        new ConnectorElementType(getAppliesToFromBpmnType(bpmnType), bpmnType))
                .collect(Collectors.toSet()) ).orElseGet(null)
```

File: element-template-generator/maven-plugin/src/main/java/io/camunda/connector/generator/ElementTemplateGeneratorMojo.java [hunk 4]
```diff
@@ -143,6 +136,28 @@ public void execute() throws MojoFailureException {
...
+      if (hybrid) {
+        fileName = fileName.replace(".json", "-hybrid.json");
+      }
+      writeElementTemplate(templates.get(0), fileName);
+    } else {
+      for (var template : templates) {
+        var fileName =
+            templateFileName == null
+                ? transformConnectorNameToTemplateFileName(template.name())
+                : templateFileName + "-" + template.elementType() + ".json";
```
Comment (markfarkas-camunda) [inline element-template-generator/maven-plugin/src/main/java/io/camunda/connector/generator/ElementTemplateGeneratorMojo.java:154]: 🙃 nitpicking: Maybe use Optional here as well, sometinhg like this:
```
Optional.ofNullable(templateFileName)
.map((fileName) -> templateFileName + "-" + template.elementType() + ".json")
.orElse(transformConnectorNameToTemplateFileName(template.name()))
```

**Review by markfarkas-camunda — COMMENTED** [review]
Looks good, just nitpicking 👍

**Review by igpetrov — APPROVED** [review]
LGTM

**Review by markfarkas-camunda — APPROVED** [review]
Looks good, thank you! 👍

**Pull request merged**